experimental/python/codegen/codegen/openapi_patch.py

```diff
@@ -0,0 +1,83 @@
+from dataclasses import replace
+
+from codegen.openapi import Schema
+
+REMOVED_FIELDS = {
+    "jobs.RunJobTask": {
+        # all params except job_parameters should be deprecated and should not be supported
+        "jar_params",
+        "notebook_params",
+        "python_params",
+        "spark_submit_params",
+        "python_named_params",
+        "sql_params",
+        "dbt_commands",
+        # except pipeline_params, that is not deprecated
+    },
+    "jobs.TriggerSettings": {
+        # Old table trigger settings name. Deprecated in favor of `table_update`
+        "table",
+    },
+    "compute.ClusterSpec": {
+        # doesn't work, openapi schema needs to be updated to be enum
+        "kind",
+    },
+    "jobs.Task": {
+        # TODO add task types as a follow-up
+        "clean_rooms_notebook_task",
+        "gen_ai_compute_task",
+    },
+    "jobs.SparkJarTask": {
+        # Deprecated. A value of `false` is no longer supported.
+        "run_as_repl",
+    },
+}
```

Comment:
I would love upstream deprecation tags for these. It won't scale to manually update these.

Reply:
All these fields are marked as deprecated, added propagation in https://github.com/databricks/cli/pull/2652

Reply:
Thanks, looks great.